common/containers/Tabs/Swap/components/CurrentRates.jsx

```diff
@@ -25,9 +26,29 @@ export default class CurrentRates extends Component {
     });
   };
 
-  render() {
-    const { ETHBTC, ETHREP, BTCETH, BTCREP } = this.props;
+  buildPairRate = (origin, destination) => {
+    const pair = origin + destination;
+    const statePair = this.state[pair + 'Amount'];
+    const propsPair = this.props[pair];
+    return !propsPair
```

Comment:
Since there is no loading state provided to the component, this implementation is a bit "dirty".

However, because rates are constantly refreshed (every 5 seconds), I thought it better to simply spin when there is no rate data instead of when a rate request is being resolved.